RxSwift/Observables/CompactMap.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  CompactMap.swift
+//  RxSwift
+//
+//  Created by Michael Long on 04/09/2019.
+//  Copyright © 2015 Krunoslav Zaher. All rights reserved.
+//
+
+extension ObservableType {
+
+    /**
+     Projects each element of an observable sequence into an optional form and filters all optional results.
+
+     Equivalent to:
+
+     func compactMap<R>(_ transform: @escaping (Self.E) throws -> R?) -> RxSwift.Observable<R> {
+        return self.map { try? transform($0) }.filter { $0 != nil }.map { $0! }
+     }
+
+     - parameter transform: A transform function to apply to each source element and which returns an element or nil.
+     - returns: An observable sequence whose elements are the result of filtering the transform function for each element of the source.
+
+     */
+    public func compactMap<R>(_ transform: @escaping (E) throws -> R?)
```

Comment:
I think this would be better off as 

```suggestion
    public func compactMap<R>(_ transform: @escaping (E) throws -> R? = { e in return e })
```

WDYT?

Reply:
(or `{ $0 }` if that works for default)

Reply:
Compiler barfs at that one. Have to add...
```
    public func compactMap<R>()
        -> Observable<R> where Self.E == R {
            return CompactMap(source: self.asObservable(), transform: { $0 })
    }
```

Reply:
I'm OK with having an extra overload, and that would also solve the test concern I mentioned.

Reply:
@freak4pc  :))
Just an opinion here, since this will allow to call it like `compactMap()`, I think for the naming, it will be confusing since is not a map anymore(because it doesn't have explicity the transform map function T -> U). Maybe in this case is more clear to define another function with a clearer naming like `compactElements()` or simply `compact()`. Not sure, maybe it makes sense :))

Reply:
I'm OK with either. TBH, even having `compactMap` only with a closure is also fine and aligns with what Swift does.

Reply:
Did a variant of this method get added that handles `.compactMap { $0 }`?